solidity/contracts/StaticOracle.sol

```diff
@@ -84,6 +85,44 @@ contract StaticOracle is IStaticOracle {
     return _quote(_baseAmount, _baseToken, _quoteToken, _pools, _period);
   }
 
+  /// @inheritdoc IStaticOracle
+  function quoteAllAvailablePoolsWithOffsettedTimePeriod(
```

Comment:
Overall looks, good, but I'm going to propose a few changes to re-use some of the code. For example:
- Make `quoteAllAvailablePoolsWithTimePeriod` call `quoteAllAvailablePoolsWithOffsettedTimePeriod` with offset = 0
- Make `quoteSpecificFeeTiersWithTimePeriod` call `quoteSpecificFeeTiersWithOffsettedTimePeriod` with offset = 0
- Make `quoteSpecificPoolsWithTimePeriod` call `quoteSpecificPoolsWithOffsettedTimePeriod` with offset = 0
- Delete `_quote`